test-data/unit/check-setattr.test

```diff
@@ -0,0 +1,12 @@
+[case testSetAttrWithProperty]
+from typing import Any
+class Foo:
+    _a = 1
+    def __setattr__(self, name: str, value: Any) -> None:
```

Comment:
Can you add these additional test cases:
* `__setattr__` inherited from a base class.
* The value argument of `__setattr__` has a specific type such as `int` (different from the type of the property). Check that assigning a non-`int` value to the attribute gets flagged by mypy.